### PR TITLE
Run tests with latest liquid/c gem

### PR DESCRIPTION
**Pull request by fw42**
Turns out https://github.com/Shopify/liquid/pull/592 is actually broken if liquid/c is present.

This PR makes it so that we run tests twice, once with and once without liquid/c enabled.

(Tests in this PR are going to fail for now. Justin is working on a fix in liquid/c.)

@pushrax @trishume

### CODE REVIEW

**Comment by pushrax** [comment]
I'm not sure about using `master` always, it means that when you check out an old liquid version the tests may fail...

Should we lock the liquid-c SHA in the gemfile?

**Comment by pushrax** [comment]
Also, it will fail on non-MRI rubies :(

**Comment by fw42** [comment]
Hmm good points

**Comment by trishume** [comment]
nice 2x2 test matrix, quadruple the error messages if you screw up something globally :smiling_imp: 

In terms of version issues, maybe gate it by an environment variable that is set in CI?

File: test/test_helper.rb [hunk 1]
```diff
@@ -15,6 +15,16 @@
 end
 Liquid::Template.error_mode = mode
 
+if ENV['LIQUID-C'] == '1'
+  puts "-- LIQUID C"
+  begin
+    require 'liquid/c'
+  rescue LoadError
+    puts "LiquidC not supported."
```
Comment (parkr) [inline test/test_helper.rb]: Nitpick, but `$stderr.puts` here?

**Comment by parkr** [comment]
:+1:

**Comment by fw42** [comment]
How about instead of using master, we use the latest released version of the liquid-c gem?

**Comment by pushrax** [comment]
@fw42 I think that still has the same problem, and also makes updating liquid and liquid-c in tandem difficult.

**Comment by fw42** [comment]
Hm I don't like the idea of hardcoding a liquid/c ref/version, we will forget to bump that.

**Pull request merged**